bazel/BUILD

```diff
@@ -164,3 +164,62 @@ alias(
         "//conditions:default": "@boringssl//:ssl",
     }),
 )
+
+# Configuration settings to make doing selects for Apple vs non-Apple platforms
+# easier. More details: https://docs.bazel.build/versions/master/configurable-attributes.html#config_settingaliasing
+config_setting(
+    name = "darwin",
+    values = {"cpu": "darwin"},
+)
+
+config_setting(
+    name = "darwin_x86_64",
+    values = {"cpu": "darwin_x86_64"},
+)
+
+config_setting(
+    name = "ios_i386",
+    values = {"cpu": "ios_i386"},
+)
+
+config_setting(
+    name = "ios_x86_64",
+    values = {"cpu": "ios_x86_64"},
+)
+
+config_setting(
+    name = "ios_armv7",
+    values = {"cpu": "ios_armv7"},
+)
+
+config_setting(
+    name = "ios_armv7s",
+    values = {"cpu": "ios_armv7s"},
+)
+
+config_setting(
+    name = "ios_arm64",
+    values = {"cpu": "ios_arm64"},
+)
+
+config_setting(
+    name = "ios_arm64e",
+    values = {"cpu": "ios_arm64e"},
+)
+
+alias(
+    name = "apple",
+    actual = select(
+        {
+            ":darwin": ":darwin",
+            ":darwin_x86_64": ":darwin_x86_64",
+            ":ios_arm64": ":ios_arm64",
+            ":ios_arm64e": ":ios_arm64e",
+            ":ios_armv7": ":ios_armv7",
+            ":ios_armv7s": ":ios_armv7s",
+            ":ios_i386": ":ios_i386",
+            ":ios_x86_64": ":ios_x86_64",
+            "//conditions:default": ":darwin",  # This is unreachable
```

Comment:
if this is unreachable, use no_match_error?
https://docs.bazel.build/versions/master/configurable-attributes.html#custom-error-messages

Reply:
This is pretty weird, which is why I linked the docs at the top which explain this further. This isn't _actually_ unreachable, _but_ since you return `:darwin` (or any other value in the dictionary here) and because if that was the current value, it would have been matched above, we know this codepath won't get chosen from the `select()` statement that calls this.

Reply:
I'm a bit confused by this explanation, can you rephrase?

Reply:
Definitely! So you can think of this as a nested select statement where this one, the inner one, returns something we know will match (the equivalent of returning true) for apple conditions. So here if `ios_armv7s` is matched, we return the same value so that the parent select statement correctly matches that case. So it's possible that none of these statements in this nested select will be true, so we need the equivalent of returning false for "none of these matched" and since select doesn't have a structure for that, we return a value we know won't be match (because if it was matched we would have returned already before)

Reply:
Thanks for explaining. I would change the comment to `# If we are on non-Apple we will fall-thru to this default. By making the value :darwin, this is guaranteed to be False and any select statements referencing //bazel:apple will not match.`. Alternatively, we could add:
```
config_setting(
  name = "false",
  values = {"_never_match": "ignore"}
)
```

or something. I bet Bazel has an in-built you can use for this somewhere, might not be documented.

Reply:
I don't see anything in src/conditions for this, but yea if you'd prefer that that's fine with me

Reply:
Updated to do that, let me know what you think!

Reply:
Looks good. Needs CI fix. @lizan WDYT?

Reply:
Seems CI is not happy with that?

Reply:
I updated to go back to the original returning `:darwin` and clarified that further in the comment. Turns out the values of the config settings are validated